core/src/components/picker-column-internal/picker-column-internal.tsx

```diff
@@ -118,6 +118,26 @@ export class PickerColumnInternal implements ComponentInterface {
     }
   }
 
+  componentDidRender() {
+    const { activeItem, el, items, isColumnVisible, value } = this;
+
+    if (isColumnVisible) {
+      if (activeItem) {
+        if (el.scrollTop !== this.getScrollTopPositionForElement(activeItem)) {
```

Comment:
Can we add this check inside of `centerPickerItemInView` instead? 

That way other code that calls `centerPickerItemInView` can benefit from this check and we only need to call `getScrollTopPositionForElement` once.

Reply:
Updated, let me know if this aligns with what you were thinking.